### PR TITLE
NPSP: Batch Data Entry Retirement

### PR DESCRIPTION
**Pull request by voduyemi**
Added Batch Data Entry Retirement Page Blocker

[W-7807537](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008Sr0oIAC/view)

Custom Labels:
bdeRetirementBlockerHeader
bdeRetirementBlockerMessage

# Critical Changes

# Changes
- **Note: this is about Batch Data Entry (BDE), an old NPSP feature. It isn't relevant for NPSP Data Importer (aka BDI), Batch Gift Entry (BGE) or Gift Entry (GE).** As of this release, Batch Data Entry is no longer supported. Most capabilities have been replaced with newer features in NPSP. If you have any in-progress batches, you can still access those records, but you can't process them using Batch Data Entry. Contact support if you need to access in-progress batches. You may want to explore Gift Entry as an alternative to Batch Data Entry. Learn more about configuring Gift Entry: https://powerofus.force.com/s/article/NPSP-Configure-Gift-Entry

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/pages/BDE_BatchEntry.page [hunk 1]
```diff
@@ -209,92 +209,96 @@
     <style>
         .ui-autocomplete-loading { background: white url('images/ui-anim_basic_16x16.gif') right center no-repeat; }
     </style>
-    
-    <apex:sectionHeader title="Batch Data Entry" subtitle="{!pageSubtitle}" />
-    <apex:messages />
+
+    <apex:outputPanel rendered="true">
+        <apex:sectionHeader title="Batch Data Entry" subtitle="{!pageSubtitle}" />
```
Comment (psadie-sfdo) [inline src/pages/BDE_BatchEntry.page:214]: Need to use the custom label for the section header title here

```suggestion
        <apex:sectionHeader title="{!$Label.stgNavBDE}" subtitle="{!pageSubtitle}" />
```

**Review by psadie-sfdo — COMMENTED** [review]
Just a couple of things, but otherwise looking pretty good to me!

File: src/lwc/utilIllustration/utilIllustration.html [hunk 2]
```diff
@@ -1049,10 +1049,26 @@
...
             </h3>
-            <p if:true={hasMessage}
-               class="slds-text-body_regular">
-               {message}
-            </p>
+            <template if:true={shouldWrapText}>
+                <lightning-layout>
+                    <lightning-layout-item size="4">
+                    </lightning-layout-item>
+                    <lightning-layout-item size="4">
```
[the remark this answers is not in the document]
Comment (voduyemi) [reply src/lwc/utilIllustration/utilIllustration.html:1056]: Good catch @psadie-sfdo I think I can use the `smallScreenSize`and `mediumScreenSize` attributes here.

[the remark this answers is not in the document]
Comment (voduyemi) [reply src/lwc/utilIllustration/utilIllustration.html]: Done!

File: src/labels/CustomLabels.labels [hunk 3]
```diff
@@ -3246,6 +3246,22 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+    <shortDescription>Header about Batch Data Entry being unsupported</shortDescription>
+    <value>Batch Data Entry Not Supported</value>
+    </labels>
+    <labels>
+    <fullName>bdeRetirementBlockerMessage</fullName>
+    <categories>Batch Data Entry</categories>
+    <language>en_US</language>
+    <protected>true</protected>
+    <shortDescription>Explanation of Batch Data Entry not being supported &amp; recommendations</shortDescription>
+    <value>As of October 1, 2020, Batch Data Entry is no longer supported. If you have any in-progress batches, you can still access those records, but you can&apos;t process them using Batch Data Entry. Contact support if you need to access in-progress batches. You may want to explore Gift Entry as an alternative to Batch Data Entry. Learn more about Gift Entry in the &lt;a href=&apos;https://powerofus.force.com/NPSP_Documentation&apos; target=&apos;_blank&apos;&gt;NPSP Documentation&lt;/a&gt; on the Power of Us Hub.</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:3263]:
```suggestion
    <value>As of October 1, 2020, Batch Data Entry is no longer supported. Most capabilities have been replaced with newer features in NPSP. If you have any in-progress batches, you can still access those records, but you can't process them using Batch Data Entry. Contact support if you need to access in-progress batches.
You may want to explore Gift Entry as an alternative to Batch Data Entry. Learn more about Gift Entry in the &lt;a href=&apos;https://powerofus.force.com/NPSP_Documentation&apos; target=&apos;_blank&apos;&gt;NPSP Documentation&lt;/a&gt; on the Power of Us Hub.</value>
```

Comment (melissabarber) [reply src/labels/CustomLabels.labels]: @voduyemi I added a suggestion that includes a second sentence as well as a hard return, but I'm sure I didn't do the carriage return correctly code-wise. Could I have you adjust that? The sentence that starts, "You may want to explore Gift Entry..." should be a second paragraph.

File: src/labels/CustomLabels.labels [hunk 4]
```diff
@@ -3246,6 +3246,23 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+    <value>Batch Data Entry Not Supported</value>
+    </labels>
+    <labels>
+    <fullName>bdeRetirementBlockerMessage</fullName>
+    <categories>Batch Data Entry</categories>
+    <language>en_US</language>
+    <protected>true</protected>
+    <shortDescription>Explanation of Batch Data Entry not being supported &amp; recommendations</shortDescription>
+    <value>As of October 1, 2020, Batch Data Entry is no longer supported. Most capabilities have been replaced with newer features in NPSP. If you have any in-progress batches, you can still access those records, but you can't process them using Batch Data Entry. Contact support if you need to access in-progress batches.
+You may want to explore Gift Entry as an alternative to Batch Data Entry. Learn more about Gift Entry in the &lt;a href=&apos;https://powerofus.force.com/NPSP_Documentation&apos; target=&apos;_blank&apos;&gt;NPSP Documentation&lt;/a&gt; on the Power of Us Hub.</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:3264]: @voduyemi looks like I left a comment that was wiped out. I added a carriage return in my suggestion, but I'm sure I didn't do it correctly code-wise. Could I have you take a look at that? What's now line 3264 should be a second paragraph in the body. Thanks!

**Comment by melissabarber** [comment]
@voduyemi @psadie-sfdo @ptewson-sfdo I updated the description of the PR to prepare it for the release notes and added a suggested change to the custom label. There's one open comment about a hard return in the custom label, but other than that, I'm signed off on this PR. Thanks!

File: src/labels/CustomLabels.labels (hunk 4 above)
Comment (voduyemi) [reply src/labels/CustomLabels.labels:3264]: Thanks Melissa, taking a look right now.

File: src/labels/CustomLabels.labels [hunk 5]
```diff
@@ -3246,6 +3246,22 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <shortDescription>Header about Batch Data Entry being unsupported</shortDescription>
+        <value>Batch Data Entry Not Supported</value>
+    </labels>
+    <labels>
+        <fullName>bdeRetirementBlockerMessage</fullName>
+        <categories>Batch Data Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Explanation of Batch Data Entry not being supported &amp; recommendations</shortDescription>
+        <value>As of October 1, 2020, Batch Data Entry is no longer supported. Most capabilities have been replaced with newer features in NPSP. If you have any in-progress batches, you can still access those records, but you can&apos;t process them using Batch Data Entry. &lt;br/&gt; Contact support if you need to access in-progress batches. You may want to explore Gift Entry as an alternative to Batch Data Entry. Learn more about Gift Entry in the &lt;a href=&apos;https://powerofus.force.com/NPSP_Documentation&apos; target=&apos;_blank&apos;&gt;NPSP Documentation&lt;/a&gt; on the Power of Us Hub.</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:3263]:
```suggestion
        <value>As of October 1, 2020, Batch Data Entry is no longer supported. Most capabilities have been replaced with newer features in NPSP. If you have any in-progress batches, you can still access those records, but you can't process them using Batch Data Entry. Contact support if you need to access in-progress batches. &lt;br/&gt; You may want to explore Gift Entry as an alternative to Batch Data Entry. Learn more about Gift Entry in the &lt;a href=&apos;https://powerofus.force.com/NPSP_Documentation&apos; target=&apos;_blank&apos;&gt;NPSP Documentation&lt;/a&gt; on the Power of Us Hub.</value>
```

Comment (melissabarber) [reply src/labels/CustomLabels.labels:3263]: @voduyemi I added one more suggested edit here to clean up a rich text apostrophe and to move the line break. Let me know if you have any questions. Thanks!

File: src/lwc/utilIllustration/utilIllustration.html [hunk 6]
```diff
@@ -1049,10 +1049,26 @@
...
-            </p>
+            <template if:true={shouldWrapText}>
+                <lightning-layout>
+                    <lightning-layout-item size="12"
+                                           large-device-size="6"
+                                           small-device-size="6"
+                                           medium-device-size="8"
+                                           class=slds-align_absolute-center>
+                        <lightning-formatted-rich-text if:true={hasMessage}
+                                                       class=slds-align_absolute-center
```
Comment (psadie-sfdo) [inline src/lwc/utilIllustration/utilIllustration.html:1060]: Minor suggestion.

```suggestion
                                                       class="slds-align_absolute-center"
```

File: src/lwc/utilIllustration/utilIllustration.html [hunk 7]
```diff
@@ -1049,10 +1049,26 @@
...
-               class="slds-text-body_regular">
-               {message}
-            </p>
+            <template if:true={shouldWrapText}>
+                <lightning-layout>
+                    <lightning-layout-item size="12"
+                                           large-device-size="6"
+                                           small-device-size="6"
+                                           medium-device-size="8"
+                                           class=slds-align_absolute-center>
```
Comment (psadie-sfdo) [inline src/lwc/utilIllustration/utilIllustration.html:1058]: Minor suggestion.
```suggestion
                                           class="slds-align_absolute-center">
```

**Review by psadie-sfdo — CHANGES_REQUESTED** [review]
Could we add a new jest test that covers the new `shouldWrapText` condition?

File: src/lwc/utilIllustration/utilIllustration.html [hunk 8]
```diff
@@ -1049,10 +1049,26 @@
...
             </h3>
-            <p if:true={hasMessage}
-               class="slds-text-body_regular">
-               {message}
-            </p>
+            <template if:true={shouldWrapText}>
+                <lightning-layout>
+                    <lightning-layout-item size="12"
+                                           large-device-size="6"
+                                           small-device-size="6"
```
Comment (psadie-sfdo) [inline src/lwc/utilIllustration/utilIllustration.html:1056]: I think it'd be great to let the layout fill and use all the available real estate for small devices.

```suggestion
                                           small-device-size="12"
```

File: src/labels/CustomLabels.labels (hunk 5 above)
Comment (voduyemi) [reply src/labels/CustomLabels.labels]: Updated!

**Review by psadie-sfdo — APPROVED** [review]
Looks good to me! Thanks Victor!

**Review by ptewson-sfdo — APPROVED** [review]
This looks good!

**Pull request merged**